### PR TITLE
Add bfloat support for torch.randn and torch.norm

### PR DESCRIPTION
**Pull request by malfet**

### CODE REVIEW

File: test/test_tensor_creation_ops.py [hunk 1]
```diff
@@ -1199,7 +1199,7 @@ def seed(generator):
             self.assertTrue((res1 < 6).all().item())
             self.assertTrue((res1 >= 0).all().item())
 
-    @dtypes(torch.half, torch.float, torch.double,
+    @dtypes(torch.half, torch.float, torch.bfloat16, torch.double,
```
Comment (walterddr) [inline test/test_tensor_creation_ops.py:1202]: should we also add `@dyptes(torch.bfloat16)` to tests in `test_linalg.py` such as `test_norm_*`?

**Review by walterddr — APPROVED** [review]
lgtm

**Review by pbelevich — APPROVED** [review]

File: aten/src/ATen/native/cpu/ReduceOpsKernel.cpp [hunk 2]
```diff
@@ -181,47 +181,47 @@ static void norm_kernel_tensor_iterator_impl(
 
 
   if (val == 0) {
-    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(kHalf, iter.dtype(), "norm_cpu", [&] {
+    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(kHalf, kBFloat16, iter.dtype(), "norm_cpu", [&] {
```
Comment (ngimel) [inline aten/src/ATen/native/cpu/ReduceOpsKernel.cpp:184]: this enables bfloat16 for cpu norm, but not for gpu norm (also, intermediate accumulation is still bfloat16, which is very imprecise, but that's a problem for another day).

File: test/test_tensor_creation_ops.py (hunk 1 above)
Comment (malfet) [reply test/test_tensor_creation_ops.py:1202]: Good idea, let me do it in followup PR

**Pull request closed without merging**